src/pages/fi-opening-balances/forms/FiOpeningBalancesForm.js

```diff
@@ -0,0 +1,212 @@
+          ...obj,
+          recordId: response.recordId 
+        })
+      } else toast.success('Record Edited Successfully')
+      setEditMode(true)
+
+      invalidate()
+    }
+  })
+
+  useEffect(() => {
+    ;(async function () {
+      try {
+        if (recordId) {
+
+          const res = await getRequest({
+            extension: FinancialRepository.FiOpeningBalance.get,
+            parameters: `_recordId=${recordId}`
+          })
+
+          formik.setValues(res.record)
+        }
+      } catch (e) { }
+    })()
+  }, [])
+
+  return (
+    <FormShell resourceId={ResourceIds.FiOpeningBalances} form={formik} maxAccess={maxAccess} editMode={editMode}>
+      <Grid container spacing={4}>
+        <Grid item xs={12}>
+          <ResourceComboBox
+            endpointId={SystemRepository.FiscalYears.qry}
+            name='fiscalYear'
+            label={labels.fiscalYear}
+            valueField='fiscalYear'
+            displayField='fiscalYear'
+            values={formik.values}
+            required
+            readOnly={editMode}
+            maxAccess={maxAccess}
+            onChange={(event, newValue) => {
+              formik.setFieldValue('fiscalYear', newValue?.fiscalYear || null)
+            }}
+            error={formik.touched.fiscalYear && Boolean(formik.errors.fiscalYear)}
+          />
+        </Grid>
+        <Grid item xs={12}>
+          <ResourceLookup
+            endpointId={FinancialRepository.Account.snapshot}
+            parameters={{
+              _countryId: formik.values.accountId,
+              _stateId: 0
+            }}
+            valueField='recordId'
+            displayField='name'
+            name='reference'
+            label={labels.accountName}
+            required
+            readOnly={editMode}
+            form={formik}
+            firstValue={formik.values.accountRef}
+            secondValue={formik.values.accountName}
+            columnsInDropDown={[
+              { key: 'reference', value: 'Reference' },
+              { key: 'name', value: 'Name' }
+            ]}
+            onChange={(event, newValue) => {
+              if (newValue) {
+                formik.setFieldValue('accountId', newValue?.recordId)
+                formik.setFieldValue('accountRef', newValue?.reference)
+                formik.setFieldValue('accountName', newValue?.name)
+              } else {
+                formik.setFieldValue('accountId', '')
+                formik.setFieldValue('accountRef', null)
+                formik.setFieldValue('accountName', null)
+              }
+            }}
+            errorCheck={'accountId'}
+            maxAccess={maxAccess}
+          />
+        </Grid>
+        <Grid item xs={12}>
+          <ResourceComboBox
+            endpointId={SystemRepository.Currency.qry}
+            name='currencyId'
+            label={labels.currencyName}
+            valueField='recordId'
+            displayField={['name','reference','flName']}
```

Comment:
sort them in the same way in the display field and columns in dropdown
reference then name then fl name